socket/sc_sock.c

```diff
@@ -972,7 +978,7 @@ int sc_sock_pipe_read(struct sc_sock_pipe *p, void *data, unsigned int len)
 	int rc;
 
 	rc = recv(p->fds[0], (char *) data, len, 0);
-	if (rc == SOCKET_ERROR || (unsigned int) rc != len) {
+	if (rc == SOCKET_ERROR) {
```

Comment:
I don't have Windows machine atm but I guess `SOCKET_ERROR` is `-1`. Just wondering if we really return negative value on `SOCKET_ERROR`.  Windows return codes are mess, some error codes are zero or positive :)

Reply:
Yes, it is `-1` : https://learn.microsoft.com/en-us/windows/win32/winsock/handling-winsock-errors